tests/test_genetic_value.py

```diff
@@ -0,0 +1,876 @@
+        L += 1
+    tables.sequence_length = L
+    tables.sort()
+    tables.simplify()
+    return tables.tree_sequence()
+
+
+class Test_output:
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    def test_output_single(self, num_ind):
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        ts = msprime.sim_ancestry(num_ind, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        trait_df = tstrait.sim_trait(
+            ts=ts,
+            num_causal=5,
+            model=model,
+            alpha=0,
+            random_seed=1,
+        )
+        genetic_df = tstrait.calculate_genetic_value(ts, trait_df)
+
+        assert len(genetic_df) == num_ind
+        assert genetic_df.shape[1] == 3
+        assert np.array_equal(genetic_df["individual_id"], np.arange(num_ind))
+        assert len(genetic_df["genetic_value"]) == num_ind
+        assert np.array_equal(genetic_df["trait_id"], np.zeros(num_ind))
+
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    def test_output_multiple(self, num_ind):
+        mean = np.array([0, 0, 0])
+        var = np.array([1, 9, 16])
+        cor = np.array([[1, 0.1, 0.2], [0.1, 1, 0.3], [0.2, 0.3, 1]])
+        model = tstrait.trait_model(
+            distribution="multi_normal", mean=mean, var=var, cor=cor
+        )
+        ts = msprime.sim_ancestry(num_ind, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        trait_df = tstrait.sim_trait(
+            ts=ts,
+            num_causal=5,
+            model=model,
+            alpha=0,
+            random_seed=1,
+        )
+        genetic_df = tstrait.calculate_genetic_value(ts, trait_df)
+
+        assert len(genetic_df) == num_ind * 3
+        assert genetic_df.shape[1] == 3
+        assert len(genetic_df["individual_id"]) == num_ind * 3
+        assert len(genetic_df["genetic_value"]) == num_ind * 3
+        assert len(genetic_df["trait_id"]) == num_ind * 3
+
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    def test_binary_mutation_model(self, num_ind):
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        ts = msprime.sim_ancestry(num_ind, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1, model="binary")
+        trait_df = tstrait.sim_trait(
+            ts=ts,
+            num_causal=5,
+            model=model,
+            alpha=0,
+            random_seed=1,
+        )
+        genetic_df = tstrait.calculate_genetic_value(ts, trait_df)
+
+        assert len(genetic_df) == num_ind
+        assert genetic_df.shape[1] == 3
+        assert np.array_equal(genetic_df["individual_id"], np.arange(num_ind))
+        assert len(genetic_df["genetic_value"]) == num_ind
+        assert np.array_equal(genetic_df["trait_id"], np.zeros(num_ind))
```

Comment:
is this equivalent to `np.all(genetic_df["trait_id"]==0)`? Not needed to allocate an entire array in memory to simply check that all entries of another array are zero.